misa-asp/src/components/AdminPage/AdminComponent.vue

```diff
@@ -0,0 +1,378 @@
+          </button>
+        </div>
+      </div>
+      <div class="search-create-container">
+        <div class="search-box">
+          <input
+            type="text"
+            v-model="searchQuery"
+            @input="searchUsers"
+            placeholder="Tìm Kiếm"
+          />
+        </div>
+        <div class="create-user">
+          <button class="create-button" @click="createUser">
+            <i class="fa fa-user-plus"></i> Tạo mới User
+          </button>
+        </div>
+      </div>
+      <table class="user-table">
+        <thead>
+          <tr>
+            <th>Họ Tên</th>
+            <th>Email</th>
+            <th>Số điện thoại</th>
+            <th>Role</th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr v-for="user in users" :key="user.id">
+            <td>{{ user.firstName }} {{ user.lastName }}</td>
+            <td>{{ user.email }}</td>
+            <td>{{ user.phoneNumber }}</td>
+            <td>{{ user.roleName }}</td>
+            <td>
+              <button @click="editUser(user.id)">Sửa</button>
+              <button @click="deleteUser(user.id)">Xóa</button>
+            </td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+  </div>
+</template>
+
+<script>
+import { deleteUserById, fetchProtectedData } from "../../api/account";
+
+export default {
+  name: "AdminComponent",
+  data() {
+    return {
+      users: [],
+      searchQuery: "",
+      userName: localStorage.getItem("lastName") || "",
+      userAvatar:
+        "https://static.vecteezy.com/system/resources/thumbnails/007/407/996/small/user-icon-person-icon-client-symbol-login-head-sign-icon-design-vector.jpg", // Placeholder avatar, you can replace with actual URL
+    };
+  },
+  async created() {
+    await this.loadUsers();
+  },
+  methods: {
+    async loadUsers() {
+      try {
+        const response = await fetchProtectedData();
+        this.users = response;
+      } catch (error) {
+        console.error("Error fetching users:", error);
+        alert(
+          "Failed to fetch user data: " +
+            (error.response ? error.response.data.message : error.message)
+        );
+      }
+    },
+    searchUsers() {
+      // Implement search functionality if needed
+    },
+    createUser() {
+      this.$router.push({ path: `/create-user` });
+    },
+    editUser(userId) {
+      this.$router.push({ path: `/edit-user/${userId}` });
+    },
+    async deleteUser(id) {
+      if (confirm("Are you sure you want to delete this user?")) {
+        try {
+          await deleteUserById(id);
+          this.users = this.users.filter((user) => user.id !== id);
+          alert("User deleted successfully!");
+        } catch (error) {
+          console.error("Error deleting user:", error);
+          alert("Failed to delete user: " + error.message);
+        }
+      }
+    },
+    logout() {
+      // Xóa role và lastName khỏi localStorage
```

Comment:
logout chỉ đơn giản xóa localStorage??? nếu như user bị lộ token thì có nghĩa máy tính của người đăng xuất k còn nhưng hacker vẫn truy cập được à?
Gọi xuống BE để xóa token